packages/builder/src/helpers/screen.ts

```diff
@@ -0,0 +1,30 @@
+import { Component, Screen, ScreenProps } from "@budibase/types"
+import clientManifest from "@budibase/client/manifest.json"
+
+export function findComponentsBySettingsType(screen: Screen, type: string) {
+  const result: Component[] = []
+  function recurseFieldComponentsInChildren(
+    component: ScreenProps,
+    type: string
+  ) {
+    if (!component) {
+      return
+    }
+
+    const componentType = component._component.split("/").slice(-1)[0]
+    const definition =
+      clientManifest[componentType as keyof typeof clientManifest]
+    if (
+      "settings" in definition &&
+      definition.settings.some((s: any) => s.type === type)
+    ) {
+      result.push(component)
+    }
+    component._children?.forEach(child => {
+      recurseFieldComponentsInChildren(child, type)
+    })
+  }
+
+  recurseFieldComponentsInChildren(screen?.props, type)
+  return result
+}
```

Comment:
This doesn't account for custom components, which are stored separately, but that's probably fine. Just as long as it's a known limitation

Reply:
I will create a ticket for a "nice to have", and to tackle this if we have some extra time